src/hotspot/share/opto/compile.cpp

```diff
@@ -3403,6 +3403,27 @@ void Compile::final_graph_reshaping_main_switch(Node* n, Final_Reshape_Counts& f
     break;
   }
 
+  case Op_ReverseV: {
+    if (n->in(1)->Opcode() == Op_ReverseV) {
+      if (n->is_predicated_vector() && n->in(1)->is_predicated_vector() &&
+          n->in(2) == n->in(1)->in(2)) {
+        n->subsume_by(n->in(1)->in(1), this);
+      } else {
+        bool is_user_blend = false;
+        for (DUIterator_Fast imax, i = n->fast_outs(imax); i < imax; i++) {
+          if (n->fast_out(i)->Opcode() == Op_VectorBlend) {
+            is_user_blend = true;
+            break;
+          }
+        }
+        if (!is_user_blend) {
+          n->subsume_by(n->in(1)->in(1), this);
+        }
+      }
+    }
+    break;
+  }
```

Comment:
The mid-end part looks good to me.  Thanks! BTW, may I ask why you add this transformation here while not in the `ReverseVNode::Idea()` ? Thanks so much！

Reply:
Blend IR node is created in a subsequent step after ReverseV node creation and are not visible to during eager identity transformation of ReverseV;   Final graph reshaping gives us opportunity to perform point transformation on graph patterns which cannot be easily handled through gvn based transformations.

Reply:
I see. Thanks for your explanation!